meilisearch/index.py

```diff
@@ -243,13 +245,16 @@ def search(self, query, opt_params=None):
         results: `dict`
             Dictionnary with hits, offset, limit, processingTime and initial query
         """
-        if opt_params is None:
-            opt_params = {}
-        search_param = {'q': query}
+        # Query parameters parsing
         for key in opt_params:
-            if isinstance(opt_params[key], list):
+            if 'facetsDistribution' in opt_params.keys() or 'facetFilters' in opt_params.keys():
```

Comment:
```suggestion
            if key == 'facetsDistribution' or key == 'facetFilters':
```

Reply:
I don't understand why you would use the condition 
`if 'facetsDistribution' in opt_params.keys() or 'facetFilters' in opt_params.keys():`

If it is true, aren't you applying json.dumps() to every single key and not only to those related to faceting? I think if you are already iterating on the keys, you can check if a key belongs to faceting, and just in that case apply json.dumps()

I am not sure about my suggestion, I ddin't test it as I'm traveilng, is more an idea that a tested solution

Reply:
In case you have `facetsDistibution` AND `attributesToRetrieve` I don't think this would work (worth doing a test?)

Reply:
Your commit suggestion was exactly what I wanted to do! That's a mistake I did!! Thanks for noticing it Sam!! 😄

> (worth doing a test?)

Even if it had worked, it's a smart move to add a test on this part because both of these attributes are concerned by 2 different "branches" of the `if` condition!
Test added!!